scripts/run-examples.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+# ===----------------------------------------------------------------------=== #
+# Copyright (c) 2024, Modular Inc. All rights reserved.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions:
+# https://llvm.org/LICENSE.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===----------------------------------------------------------------------=== #
+
+set -euo pipefail
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+REPO_ROOT="${SCRIPT_DIR}"/../
+EXAMPLES_DIR="${REPO_ROOT}"/examples
+
+echo "Creating build directory"
+BUILD_DIR="${REPO_ROOT}"/build
+mkdir -p "${BUILD_DIR}"
+
+# Change into the examples dir where lit.cfg.py is
+# and then invoke lit to run the examples.
+(cd "${EXAMPLES_DIR}" && lit -sv .)
```

Comment:
I'm surprised one can't just invoke `lit -sv "${EXAMPLES_DIR}"` here. I sort of thought that was the point of setting the `exec_root` in the lit config -- to have lit execute from a specific location, regardless of the location from which the user invoked `lit`.

Reply:
Thanks for keeping me accountable here.  I had to tweak the `test_source_root` as it was only using `Path.cwd()`, but it really needs to account for where it may be run as seen with:

```
config.test_source_root = Path(__file__).parent.resolve()
```

This makes it so test discovery works as expected and one doesn't need to invoke `lit` only from the examples directory (and thus `cd` there first).  It just works as you'd expect now 🎉